### DISCUSSION TITLE
minio scale question: how scale a bucket capacity?

### DISCUSSION THREAD

**Discussion by lddlww** [minio/minio#16376] [Q&A] [answered]
**background：**
my minio cluster contains 4 server nodes and 1 disk per server, suppose total capacity exclude parity capacity is 1GiB 
ps：that's mean this minio cluster consist of only one ```ES``` which contains 2 data drives and 2 parity drives

then i created one bucket named mybucket，and put some large objects to mybucket，sooner as mybucket total size is close to 1GiB putting objects return fails; next，i added 1 server node to  minio cluster 

**questions:：after added new server node**
1、the above cluster ```ES``` scale to 3 data drives and keep 2 parity drives, is it right? 
2、at this time，putting objects to mybucket could success？
3、i wonder to know if mybucket could spread across two or more ```ES```?

thanks in advance

**Comment by klauspost** [comment, marked as answer]
https://min.io/docs/minio/linux/operations/install-deploy-manage/expand-minio-deployment.html has our recommendations.

> The new server pool does not need to be substantially similar in hardware and software configuration to any existing server pool, though this may allow for simplified cluster management and more predictable performance across pools.

> See [Production Hardware Requirements](https://min.io/docs/minio/linux/operations/checklists/hardware.html#deploy-minio-distributed-recommendations) for more guidance on selecting hardware for MinIO deployments.

**Reply by lddlww** [reply]
>MinIO supports expanding an existing distributed deployment by adding a new [Server Pool](https://min.io/docs/minio/linux/operations/concepts.html#minio-intro-server-pool).

so scaling minio cluster only support server pool added and single server node is not allow
and a bucket in ES which is close to total capacity can't scale?

or one minio bucket data could spread across many server pools？

**Reply by klauspost** [reply]
> so scaling minio cluster only support server pool added and single server node is not allow

Correct.

> and a bucket in ES which is close to total capacity can't scale?
> or one minio bucket data could spread across many server pools？

Data will be spread across pools on upload.

**Reply by lddlww** [reply]
thank you very much
